src/eng_back.c

```diff
@@ -241,7 +241,13 @@ ENGINE_CTX *ctx_new()
 	ctx->rwlock = CRYPTO_get_dynlock_create_callback() ?
 		CRYPTO_get_new_dynlockid() : 0;
 #endif
-
+/*
+ * OpenSSL Version prior to 1.1.0 are known to work
+ * with random.
+ */
+#if OPENSSL_VERSION_NUMBER < 0x101010000
+	ctx->enable_rand = 1;
+#endif
```

Comment:
Shouldn't `ctx->enable_rand` be explicitly set to zero otherwise?

Reply:
malloc + memset to zero, see: https://github.com/OpenSC/libp11/blob/master/src/eng_back.c#L221-L224

Reply:
Explicit is better than implicit.  Code clarity is more important than a tiny performance optimization

Reply:
Can't you see that setting `ctx->enable_rand = 0;` on `#else` would improve code readability?  @mouse07410 is right here.

Reply:
No, especially when the initialization routine is inconsistent in it's practice and about 20 lines before is a memset to zero.
Why memset the memory to write a zero to it? Why malloc + memset when _zalloc is available?

The block from 227-236 is incredibly convoluted to read. Conditional code is harder to read, especially runtime conditional code mixed with compile time conditional code. That should probably be moved to it's own routine.

With that said will I do it, sure.